mapie/metrics.py

```diff
@@ -737,3 +737,255 @@ def hsic(
+    0.8
+    """
+
+    # Ensure inputs are NumPy arrays for consistent operations
+    y_true = np.asarray(y_true)
+    y_pred_low = np.asarray(y_pred_low)
+    y_pred_up = np.asarray(y_pred_up)
+
+    in_the_range = np.sum((np.greater_equal(y_true, y_pred_low))
+                          & (np.less_equal(y_true, y_pred_up)))
+
+    coverage = in_the_range / np.prod(y_true.shape)
+
+    return float(coverage)
+
+
+def _pinaw(
+    y_true: ArrayLike,
+    y_pred_low: ArrayLike,
+    y_pred_up: ArrayLike
+) -> float:
+    """
+    Calculate the Prediction Interval Normalized Average Width (PINAW).
+
+    The Prediction Interval Normalized Average Width (PINAW) is a measure
+    used to evaluate the average width of prediction intervals (PIs) in
+    relation to the range of the true labels.
+
+
+    [5] Vilde Jensen, Filippo Maria Bianchi, Stian Norman Anfinsen (2022).
+    Ensemble Conformalized Quantile Regression for Probabilistic Time Series
+    Forecasting.
+
+    Parameters
+    ----------
+    y_true : ArrayLike
+        Array of true labels.
+    y_pred_low : ArrayLike
+        Array of lower bounds of prediction intervals.
+    y_pred_up : ArrayLike
+        Array of upper bounds of prediction intervals.
+
+    Returns
+    -------
+    float
+        Prediction Interval Normalized Average Width (PINAW).
+
+    Notes
+    -----
+    The PINAW is calculated as follows:
+    1. Calculate the average width of prediction intervals by taking the mean
+       of the absolute differences between the upper and lower bounds:
+       avg_length.
+    2. Normalize avg_length by dividing it by the range of the true labels:
+       (y_true.max() - y_true.min()).
+    3. The resulting value is the PINAW.
+
+    Examples
+    --------
+    >>> y_true = [5, 7.5, 9.5, 10.5, 12.5]
+    >>> y_pred_low = [4, 6, 9, 8.5, 10.5]
+    >>> y_pred_up = [6, 9, 10, 12.5, 12]
+    >>> y_true = np.array(y_true)
+    >>> y_pred_low = np.array(y_pred_low)
+    >>> y_pred_up = np.array(y_pred_up)
+    >>> print(np.round(_pinaw(y_true, y_pred_low, y_pred_up),2))
+    0.31
+    """
+    # Convert y_true to a NumPy array of floats
+    y_true = np.array(y_true, dtype=float)
+
+    avg_length = np.mean(np.abs(np.subtract(y_pred_up, y_pred_low)))
+    avg_length = avg_length / (np.subtract(float(y_true.max()),
+                                           float(y_true.min())))
+
+    return float(avg_length)
+
+
+def cwc(
+    y_true: ArrayLike,
+    y_pred_low: ArrayLike,
+    y_pred_up: ArrayLike,
+    eta: float,
+    mu: float
+) -> float:
+    """
+    Coverage width-based criterion obtained by the prediction intervals.
+
+    The effective coverage score is a criterion used to evaluate the quality
+    of prediction intervals (PIs) based on their coverage and width.
+
+
+    [5] Vilde Jensen, Filippo Maria Bianchi, Stian Norman Anfinsen (2022).
+    Ensemble Conformalized Quantile Regression for Probabilistic Time Series
+    Forecasting.
```

Comment:
@QMaphan, are you sure that this paper is the first one that proposes this metric?